sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1542,6 +1545,8 @@ class ResourceHandle(object):
         - (required) Path to a cluster.yaml file.
         - (optional) A cached head node public IP.  Filled in after a
             successful provision().
+        - (optional) A cached stable list of internal IPs for all nodes
+            in a cluster. Filled in after a successful task execution.
```

Comment:
As mentioned above, can we cache both `external` and `internal` ips?

It can be used to speed up the places we call `get_node_ips` in the `cloud_vm_ray_backend`. With that said, I would prefer to add another argument called `use_cached_ips: bool` to the `backend_utils.get_node_ips`, so we can extend the following lines to use the cached ips.
https://github.com/skypilot-org/skypilot/blob/b2d3555af1f4b5aba53cc0c04c39ccd44fa26bc1/sky/backends/backend_utils.py#L1152-L1155

Reply:
Since the `get_stable_cluster_ips()` function now calls `get_node_ips()`, what do you think about adding the `use_cached_ips` argument to the `get_stable_cluster_ips()` function and replacing all `get_node_ips()` calls with `get_stable_cluster_ips()` calls instead? This is so that `get_stable_cluster_ips()` is now our "top-level" function. I think this makes more sense because otherwise we would have both functions calling each other in different cases.

Reply:
Good point! I think it should be fine to replace the `get_node_ips` with `get_stable_cluster_ips`.